src/Raven.Server/Utils/ReplicationUtils.cs

```diff
@@ -4,22 +4,179 @@
+                        LastIndexTransformerEtag = incomingHandler.LastIndexOrTransformerEtag,
+                        LastHeartbeatTicks = incomingHandler.LastHeartbeatTicks
+                    });
+            }
+
+            foreach (var destination in replicationDocument.Destinations)
+            {
+                OutgoingReplicationHandler outgoingHandler;
+                DocumentReplicationLoader.ConnectionFailureInfo connectionFailureInfo;
+
+                if (TryGetActiveDestination(
+                    destination,
+                    replicationLoader.OutgoingHandlers,
+                    out outgoingHandler))
+                {
+                    var changeVector = outgoingHandler._database.DocumentsStorage.GetDatabaseChangeVector(context);
+                    var globalChangeVector = ConvertChangeVectorToDictionary(changeVector);
+                    topologyInfo.Outgoing.Add(
+                        new ActiveNodeStatus
+                        {
+                            DbId = outgoingHandler.DestinationDbId,
+                            IsOnline = true,
+                            LastDocumentEtag = outgoingHandler._lastSentDocumentEtag,
+                            LastIndexTransformerEtag = outgoingHandler._lastSentIndexOrTransformerEtag,
+                            LastHeartbeatTicks = outgoingHandler.LastHeartbeatTicks,
+                            GlobalChangeVector = globalChangeVector,
+                            NodeStatus = ActiveNodeStatus.Status.Online
+                        });
+
+                    activeDestinations.Add(destination);
+                }
+                else if (replicationLoader.OutgoingFailureInfo.TryGetValue(destination, out connectionFailureInfo))
+                {
+                    topologyInfo.Outgoing.Add(
+                        new ActiveNodeStatus
+                        {
+                            DbId = connectionFailureInfo.DestinationDbId,
+                            IsOnline = false,
+                            LastDocumentEtag = connectionFailureInfo.LastSentDocumentEtag,
+                            LastIndexTransformerEtag = connectionFailureInfo.LastSentIndexOrTransformerEtag,
+                            LastHeartbeatTicks = connectionFailureInfo.LastHeartbeatTicks,
+                            GlobalChangeVector =
+                                ConvertChangeVectorToDictionary(connectionFailureInfo.GlobalChangeVector),
+                            NodeStatus = ActiveNodeStatus.Status.Online
+                        });
+                }
+                else
+                {
+                    Exception isAliveCheckException = null;
+                    try
+                    {
+                        var isAliveTask = GetTcpInfoAsync(
+                            context, destination.Url,
+                            destination.Database, destination.ApiKey);
+
+                        isAliveTask.Wait(database.DatabaseShutdown);
```

Comment:
You are waiting here _synchronously_ , why?
This might take a very long time, and you have _already_ tried connecting to them and failed.